### ISSUE TITLE
[Bug]: New user via external login can delete their IdP link and make their account inaccessible

### ISSUE THREAD

**Issue by kraney** [zitadel/zitadel#6081]
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

Self-hosted

### Describe the problem caused by this bug

I was testing the UX for an unprivileged user. I have an organization set up where username/password login is disallowed, and it has a single external IdP which is the only way to sign on. Registration is allowed and accounts are created automatically.

After signing in a new user so the registration happened, I browsed around the console looking for ways an unprivileged user could get themselves in trouble.

In the "Identity Providers" panel, there is a delete control for the IdP. The user is able to delete this, and if they do they are screwed. There's no way to re-link the external login to the existing account, but you also can't create a new account since the account already exists.

Really, the user should not be able to perform this action. Also, since they can't, the delete button should not appear. (This is fairly common - for example in Authorizations a delete button is shown for a role, but in this case clicking it just returns an error due to lack of privileges)

Secondarily, the dialog for this does not have proper localization.

### To reproduce

1. Create an organization that is authenticated only by external IdP
2. Create a new account within that organization (auto-create on login)
3. Navigate to Identity Providers panel
4. Shoot self in foot.

### Screenshots

<img width="1824" alt="image" src="https://github.com/zitadel/zitadel/assets/5760081/1f944e9e-33a3-4edb-94cd-f99de861dd01">

### Expected behavior

* User is unable to perform a fatal action, especially given lack of privileges.
* Less good alternative - there is a way to recover such an account / re-link for the user.

**Comment by kraney** [comment]
Version is 2.28.1. Not sure why it shows "No response".

**Comment by fforootd** [comment]
Hm lets us look into this.

**Comment by hifabienne** [comment]
Hei @kraney 
Thanks for opening up the report. With our users profile page we want to give a standard ui that fits multiple use cases. If you have other needs than that we recommend building your own ui.

There are use cases where you want to enable your users to manage their own identity provider links. Imaging you have a SaaS application for End users like an online shop, spotify or netflix. In that case you want your users to be able to link their Google account or to remove it if they don't have it anymore.

So I do not agree that this should not be possible.
Where I do agree is that the self recovery part on the side of ZITADEL is not yet good enough. The users should either not be able to delete the "last" authentication possibility or should be able to recover the account in that case.

**Comment by ahmednfwela** [comment]
There is a similar bug to this that I have hit on a production database.
disabling "Username Password allowed" will disallow existing users from logging in with their usernames and passwords, so if the admin account doesn't have an external Idp, you are basically locked out of the instance for good.

**Comment by hifabienne** [comment]
> > There is a similar bug to this that I have hit on a production database. disabling "Username Password allowed" will disallow existing users from logging in with their usernames and passwords, so if the admin account doesn't have an external Idp, you are basically locked out of the instance for good.
> 
> Yeah that is a slight annoyance 😁
> 
> @hifabienne I think we need to come up with either a safeguard for this or a cli that can resolve this problem, what do you think?

Yes, I think the right way would be to have the possibility to login one more time with the username password and to setup the external provider.

**Comment by livio-a** [comment]
I've checked the options and TBD in the meeting today:
1. Prevent remove:
To discuss where to prevent (Console or backend) and check if a password is set (and allowed to be used / local auth enabled)
2. Recovery:
Would be possible with the new APIs:
    - RetrieveIdentityProviderIntent provides all the infos
    - Check if user exists (loginname / email query on ListUsers)
    - Use the infos to AddIDPLink on existing user

**Comment by livio-a** [comment]
As discussed with @muhlemmer , we propose to add a warning in Console, but not preventing the delete itself.

**Comment by elinashoko** [comment]
Thanks for filing this issue. As part of an ongoing backlog cleanup, we're closing issues that have had no activity in over 12 months.

If this is still relevant, let us know in a comment (or reopen the issue) and we'll take another look.

This issue will close in 30 days if there's no further activity.
